### ISSUE TITLE
Rule to detect invalid FSD layers (e.g. model/ in module root)

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#28]
## Problem

The plugin currently validates import directions between layers, but does not detect when an invalid layer directory is created in the module root.

For example, in a project structure like:

```
modules/
  knowledge/
    shared/
    entities/
    features/
    widgets/
    pages/
    model/        <-- invalid! "model" is a segment, not a layer
```

The `model/` directory is not a valid FSD layer - it's a segment that should exist inside a slice (e.g. `pages/sections/model/`). However, no rule currently catches this.

## Expected behavior

A rule (or option on existing rules) that validates the directory structure and reports when a non-layer directory exists at the root of a module. Valid layers should be configurable (matching the `layers` option from the plugin config).

## Current workaround

We wrote a custom ESLint rule that checks the file path:

```js
const VALID_LAYERS = new Set(['shared', 'entities', 'features', 'widgets', 'pages'])
const MODULE_PATH_REGEX = /client\/modules\/[^/]+\/([^/]+)\//

// In create():
const filename = context.filename
const match = filename.match(MODULE_PATH_REGEX)
if (match && !VALID_LAYERS.has(match[1])) {
  context.report({ ... })
}
```

This works but is fragile (hardcoded path pattern) and would be better handled by the plugin itself, which already knows the configured layers.

## Use case

This is particularly important when using AI coding assistants that may create files in incorrect locations (e.g. `modules/knowledge/model/types.ts` instead of `modules/knowledge/shared/model/types.ts`).
